overlord/overlord.go

```diff
@@ -209,7 +211,45 @@ func (o *Overlord) addManager(mgr StateManager) {
 	o.stateEng.AddManager(mgr)
 }
 
-func loadState(backend state.Backend, restartHandler restart.Handler) (*state.State, error) {
+func initStateFileLock() (*osutil.FileLock, error) {
+	lockFilePath := dirs.SnapStateLockFile
+	if err := os.MkdirAll(filepath.Dir(lockFilePath), 0755); err != nil {
+		return nil, err
+	}
+
+	return osutil.NewFileLockWithMode(lockFilePath, 0644)
+}
+
+func lockWithTimeout(l *osutil.FileLock, timeout time.Duration) error {
+	startTime := time.Now()
+	const retryInterval = time.Second
+	for {
+		err := l.TryLock()
```

Comment:
Hmm I actually liked the ticker and that we could log each attempt and know that things are happening in the background.

Reply:
But that message was anyway emitted every minute, while here we are waiting for 20 seconds at most :-)

In any case we do log before and after attempting to acquire the log, so I think we are well covered. The previous variant with the ticker made sense because we were potentially getting stuck forever, but this is not the case anymore.